mirdata/download_utils.py

```diff
@@ -246,10 +247,28 @@ def extractall_unicode(zfile, out_dir):
     for m in zfile.infolist():
         data = zfile.read(m)  # extract zipped data into memory
 
-        if m.filename.encode("cp437").decode() != m.filename.encode("utf8").decode():
-            disk_file_name = os.path.join(out_dir, m.filename.encode("cp437").decode())
+        ### get filename
+        name = m.filename
+        try:
```

Comment:
I have to admit I'm pretty confused by this code! So we're...taking the original filename, encoding it according to different schemes (lines 252-261), then detecting the encoding we just used (line 264), and then decoding it again?

I have no better suggestion... but is there not a more straightforward way?

Reply:
the original solution was this:
```
 if m.filename.encode("cp437", errors="ignore").decode(errors="ignore") != m.filename.encode("utf8", errors="ignore").decode(errors="ignore"):
            disk_file_name = os.path.join(out_dir, m.filename.encode("cp437", errors="ignore").decode(errors="ignore"))
        else:
            disk_file_name = os.path.join(out_dir, m.filename)
```
but the tests fail for this because ignoring errors in encoding leads to an incorrect encoding for irmas. 
so here:
```
        try:
            # non-utf encoding
            filename = name.encode("cp437")
        except UnicodeEncodeError:
            # utf encoding
            filename = name.encode("utf8")

        # check for irmas filename encoding
        if filename.decode(errors="ignore") != name.encode("utf8").decode():
            filename = name.encode("cp437")
```
I deal with exceptions related to irmas and good-sounds
while here:
```
        # detect encoding
        encoding = chardet.detect(filename)["encoding"]
        # decode with the encoding and ignore errors in filename
        if encoding is not None:
            filename = filename.decode(encoding, errors="ignore")
        else:
            filename = filename.decode(errors="ignore")
```
I try to deal with the other encodings

Reply:
I will run tests with a few datasets to make sure this does not break anything.

Reply:
unless we know exactly the encoding then this is difficult. I have tried to refactor this code but the tests break with any other solution.

Reply:
@rabitt , I found a more elegant solution to this problem and refactored the code.